src/native/windows/apps/prunsrv/prunsrv.c

```diff
@@ -1876,8 +1908,21 @@ void WINAPI serviceMain(DWORD argc, LPTSTR *argv)
          */
         apxLogWrite(APXLOG_MARK_DEBUG "Waiting 1 minute for all threads to exit.");
         reportServiceStatus(SERVICE_STOP_PENDING, NO_ERROR, ONE_MINUTE);
-        apxDestroyJvm(ONE_MINUTE);
-        /* if we are not using JAVA apxDestroyJvm does nothing, check the chid processes in case they hang */
+        if (!apxDestroyJvm(ONE_MINUTE)) {
+            /* if we are not using JAVA apxDestroyJvm does nothing, check the chid processes in case they hang */
+            apxLogWrite(APXLOG_MARK_DEBUG "Not using JAVA apxDestroyJvm did nothing");
+            DWORD count = SO_STOPTIMEOUT;
+            if (!count)
+                count = ONE_MINUTE_SEC;
+            do {
```

Comment:
This looks like a possible 3rd wait using the stop timeout.

Reply:
Yes we are trying to kill any processes we created in that case.
So I need to add a something to store the total time procrun has been waiting :-(
And hope that the timeout is big enough to make sure the processes cleaning is done.

Reply:
fixed